src/ocaml/merlin_specific/browse_raw.ml

```diff
@@ -894,3 +894,9 @@ let node_of_binary_part env part =
   | Partial_module_type x ->
     Module_type x
 
+let rec all_holes (env, node) =
+  let f env node acc = match node with
+    | Expression { exp_desc = Texp_hole; exp_loc; _}  -> exp_loc :: acc
+    | _ -> List.rev_append (all_holes (env, node)) acc
```

Comment:
I'm a bit surprised to see this `rev_append` inside a fold.
Why aren't we just folding down?

Reply:
`fold_node` is not really folding on all the nodes but all the "surface" nodes. To go deeper we need additional recursive calls and thus some lists concatenation.